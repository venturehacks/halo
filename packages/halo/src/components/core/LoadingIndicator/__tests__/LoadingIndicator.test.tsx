import { render } from 'enzyme';
import React from 'react';

import { LoadingIndicator } from '../index';

describe('LoadingIndicator', () => {
  test('smoke', () => {
    const component = render(<LoadingIndicator />);
    expect(component.text()).toContain('');
  });

  describe('snapshot', () => {
    test('primary', () => {
      const component = render(<LoadingIndicator variant="primary" bare />);

      expect(component).toMatchSnapshot();
    });

    test('alternate', () => {
      const component = render(<LoadingIndicator variant="alternate" bare />);

      expect(component).toMatchSnapshot();
    });

    test('default component wrapping', () => {
      // should wrap in div.component
      const component = render(<LoadingIndicator variant="alternate" />);

      expect(component).toMatchSnapshot();
    });
  });
});
