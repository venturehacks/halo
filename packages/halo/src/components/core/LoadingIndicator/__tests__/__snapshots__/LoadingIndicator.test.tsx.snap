// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator snapshot alternate 1`] = `
<span
  class="css-o6cbs8"
/>
`;

exports[`LoadingIndicator snapshot default component wrapping 1`] = `
<div
  class="component"
>
  <span
    class="css-o6cbs8"
  />
</div>
`;

exports[`LoadingIndicator snapshot primary 1`] = `
<span
  class="css-1xdhyk6"
>
  <span
    class="css-fl1se0"
  />
  <span
    class="css-1qccjm5"
  />
  <span
    class="css-do256"
  />
  <span
    class="css-1vt3jh2"
  />
  <span
    class="css-dx1ugf"
  />
</span>
`;
